cmd/test/test.go

```diff
@@ -0,0 +1,165 @@
+	"github.com/spf13/cobra"
+
+	"github.com/enterprise-contract/ec-cli/internal/applicationsnapshot"
+)
+
+const (
+	OutputAppstudio = "appstudio"
+)
+
+var TestCmd *cobra.Command
+
+func fetchTestCmd() *cobra.Command {
+	flags := struct {
+		Policy             []string
+		Data               []string
+		Namespace          []string
+		AllNamespaces      bool
+		NoFail             bool
+		FailOnWarn         bool
+		Output             string
+		NoColor            bool
+		SuppressExceptions bool
+		Trace              bool
+		JunitHideMessage   bool
+	}{}
+
+	cmd := &cobra.Command{
+		Use:   "test [flags] input",
+		Short: "A wrapper for the conftest test command",
+
+		Long: hd.Doc(`
+			The ec test command is meant to work as a drop-in replacement for conftest test.
+
+			It supports most but not all of the flags supported by Conftest. Its supports one
+			more output format than Conftest, the 'appstudio' format, which is suitable for use
+			with the standard AppStudio TEST_OUTPUT Tekton result format.
+		`),
+
+		Example: hd.Doc(`
+			The standard usage is similar to Conftest:
+
+				ec test --policy my_policies.rego --data my_data.json --namespace main input.json
+
+			Like Conftest it will traverse directories to find multiple policies and data files:
+
+				ec test --policy ./policy --data ./data --namespace main input.json
+		`),
+
+		RunE: func(cmd *cobra.Command, fileList []string) (err error) {
+			if len(fileList) < 1 {
+				return fmt.Errorf("at least one input file must be specified")
+			}
+
+			r := &runner.TestRunner{
+				Policy:             flags.Policy,
+				Data:               flags.Data,
+				Namespace:          flags.Namespace,
+				AllNamespaces:      flags.AllNamespaces,
+				NoFail:             flags.NoFail,
+				FailOnWarn:         flags.FailOnWarn,
+				Output:             flags.Output,
+				NoColor:            flags.NoColor,
+				SuppressExceptions: flags.SuppressExceptions,
+				Trace:              flags.Trace,
+			}
+
+			ctx := cmd.Context()
+			results, err := r.Run(ctx, fileList)
+			if err != nil {
+				return fmt.Errorf("running test: %w", err)
+			}
+
+			if flags.Output == OutputAppstudio {
+				// The appstudio format is unknown to Conftest so we handle it ourselves
+				report := applicationsnapshot.AppstudioReportFromCheckResults(results)
+				reportOutput, err := json.Marshal(report)
+				if err != nil {
+					return fmt.Errorf("output results: %w", err)
+				}
+				fmt.Printf("%s", reportOutput)
+
+			} else {
+				// Conftest handles the output
+				o := output.Get(r.Output, output.Options{
+					NoColor:            r.NoColor,
+					SuppressExceptions: r.SuppressExceptions,
+					Tracing:            r.Trace,
+					JUnitHideMessage:   flags.JunitHideMessage,
+				})
+
+				if err := o.Output(results); err != nil {
+					return fmt.Errorf("output results: %w", err)
+				}
+			}
+
+			exitCode := chooseExitCode(r.NoFail, r.FailOnWarn, results)
+			os.Exit(exitCode)
+			return
```

Comment:
Consider something like this:
```go
return chooseExitError(r.NoFail, r.FailOnWarn, results)
```

Reply:
Not applicable now that I'm trying to use conftest code as-is.